sklearn/preprocessing/data.py

```diff
@@ -1325,6 +1325,20 @@ def normalize(X, norm='l2', axis=1, copy=True, return_norm=False):
     return_norm : boolean, default False
         whether to return the computed norms
 
+    Returns
+    -------
+    X : {array-like, sparse matrix}, shape [n_samples, n_features]
+        Normalized input X.
+
+    norms :
+        When X is sparse, an NotImplementedError will be raised for
+        norm 'l1' or 'l2'; an array of maximum elements along given axis
+        for norm 'max'.
+
+        When X is dense, an array of norms along given axis for dense matrix X.
```

Comment:
Put this first. Drop "when X is dense" and just specify the sparse error case afterwards